src/nvim/lua/executor.c

```diff
@@ -1152,7 +1152,12 @@ int nlua_call(lua_State *lstate)
   size_t name_len;
   const char *name = luaL_checklstring(lstate, 1, &name_len);
   if (!nlua_is_deferred_safe() && !viml_func_is_fast(name)) {
-    return luaL_error(lstate, e_fast_api_disabled, "Vimscript function");
+    size_t length = strlen(name) + sizeof("Vimscript function \"\"") + 1;
+    char *str = xmalloc(length);
+    vim_snprintf(str, length, "Vimscript function \"%s\"", name);
+    int ret = luaL_error(lstate, e_fast_api_disabled, str);
+    xfree(str);
```

Comment:
can we use `IObuff` or `NameBuff` instead of allocating? Will need to truncate if the string is bigger than IOSIZE, but that doesn't seem likely here.

sth like this (untested):

```suggestion
    size_t length = MIN(strlen(name), 100) + sizeof("Vimscript function \"\"") + 1;
    vim_snprintf(IObuff, length, "Vimscript function \"%s\"", name);
    int ret = luaL_error(lstate, e_fast_api_disabled, IObuff);
```

Reply:
oops, `luaL_error` can trigger side effects, so maybe an allocation is needed.

however (off topic) we could avoid an allocation here, since `lua_pushlstring` doesn't have side effects: https://github.com/neovim/neovim/blob/76eca621374b7d19604c820d8ebaefb184de1fc9/src/nvim/lua/executor.c#L1031-L1033

cc @bfredl to confirm

Reply:
correct. pushlstring allocates a copy inside the gc heap anyway.